### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by AndrewTehan**

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+require 'rails_helper'
+
+RSpec.describe UsersController, type: :controller do
+  before :each do
+    user.confirm
```
Comment (longbeardman) [inline spec/requests/users_spec.rb:5]: seems to be redundant

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    it 'renders services' do
+      get :services, params: params
+      expect(response).to render_template('services')
+    end
+  end
+
+  describe 'DELETE user' do
+    it 'has a 200 status code' do
+      delete :destroy, params: params
+      expect(response.status).to eq(302)
```
Comment (longbeardman) [inline spec/requests/users_spec.rb:51]: check that user is removed from db

File: spec/requests/users_spec.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    it 'has a 302 status code' do
+      get :edit, params: params
+      expect(response.status).to eq(200)
+    end
+
+    it 'renders edit' do
+      get :edit, params: params
+      expect(response).to render_template('edit')
+    end
+  end
```
Comment (longbeardman) [inline spec/requests/users_spec.rb:65]: missing test cases (CRUD related)
1) create user
2) update user

File: spec/requests/visits_spec.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    end
+
+    it 'renders new' do
+      get :new, params: params
+      expect(response).to render_template('new')
+    end
+  end
+
+  describe 'create visit' do
+    let(:visit) { FactoryBot.create(:visit, client: user) }
```
Comment (longbeardman) [inline spec/requests/visits_spec.rb:50]: seems to redundant

File: spec/requests/visits_spec.rb [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                              service_id: visit.service.id,
+                              master_id: visit.master.id,
+                              addition: visit.addition,
+                              date: visit.date,
+                              service_visit_attributes: { service_id: visit.service.id } } }
+    end
+
+    it 'has a 302 status code' do
+      post :create, params: params
+      expect(response.status).to eq(302)
```
Comment (longbeardman) [inline spec/requests/visits_spec.rb:62]: test case doesn't check that visit created

File: spec/requests/visits_spec.rb [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+    it 'renders new' do
+      get :new, params: params
+      expect(response).to render_template('new')
+    end
+  end
+
+  describe 'create visit' do
+    let(:visit) { FactoryBot.create(:visit, client: user) }
+    let(:params) do { locale: I18n.locale, client_id: visit.client.id,
```
Comment (longbeardman) [inline spec/requests/visits_spec.rb:51]: create `client`, `master` and `service` in `let`
```
let(:client) { create(:client) }
let(:service) { create(:service) }
let(:master) { create(:master, service_id: service.id) }
```

File: spec/requests/visits_spec.rb [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+                      visit: { client_id: visit.client.id,
```
Comment (longbeardman) [inline spec/requests/visits_spec.rb:52]: params may be hardcored

File: spec/requests/visits_spec.rb (hunk 5 above)
Comment (longbeardman) [reply spec/requests/visits_spec.rb:62]: this test case shows that it's possible to create visits for the same date time.
this should be validated

File: spec/requests/visits_spec.rb [hunk 8]
```diff
@@ -0,0 +1,76 @@
+require 'rails_helper'
```
Comment (longbeardman) [inline spec/requests/visits_spec.rb:1]: run rubocop

File: spec/requests/visits_spec.rb [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    end
+  end
+
+  describe 'DELETE visit' do
+    let(:visit) { FactoryBot.create(:visit, client: user) }
+    let(:params) do { locale: I18n.locale,  id: visit.id, client_id: visit.client.id }
+    end
+
+    it 'has a 302 status code' do
+      delete :destroy, params: params
```
Comment (longbeardman) [inline spec/requests/visits_spec.rb:72]: check if visit removed from db

File: spec/requests/visits_spec.rb [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+      expect(response.status).to eq(302)
+    end
+  end
```
Comment (longbeardman) [inline spec/requests/visits_spec.rb:75]: missing test case for disabling visit editing after visit got confirmed

**Review by longbeardman — CHANGES_REQUESTED** [review]
minitest files (`test` folder) should be removed

File: spec/requests/visits_spec.rb (hunk 10 above)
Comment (AndrewTehan) [reply spec/requests/visits_spec.rb:97]: не получилось написать тест. тесткейс закомментирован

**Pull request merged**